### DISCUSSION TITLE
Wrong reading(?)

### DISCUSSION THREAD

**Discussion by ZeyadDP** [gijzelaerr/python-snap7#336] [Q&A] [answered]
Hello,
I am trying to read data from a PLC, but the readings are completely wrong. For example, I had a value of **27.31482** on the PLC, but what I got was **bytearray(b'A')**. This is a recurring problem, with none of the values I read being read properly.
My code:
```
import snap7 as snap

client = snap.client.Client()
client.connect('XXX.XXX.X.X', 0, 2)


intDB = []
db = client.db_read(20)

for i in range(0, len(db)):
    causSodaC = client.db_read(20, i, 1)

    causSodaC.extend(bytearray(4 - len(causSodaC)))

    realReading = snap.util.get_real(causSodaC, 0)

    array = [realReading]
    intDB.append(array)

print(intDB)
```

**Comment by swamper123** [comment]
What do you expect with:

`db = client.db_read(20)` ? The size of the DB? 
And if this is possible, is this safe/common use? @gijzelaerr 


`causSodaC = client.db_read(20, i, 1)`
Are your single values 1 byte large?

**Reply by ZeyadDP** [reply]
1. No, I expect, and get, the entirety of the data block.
2. Hmm, this might actually be the issue! I will make sure to test it as soon as I can.

**Reply by ZeyadDP** [reply]
Thanks, it worked! The sizes of the were 4 bytes long.

**Comment by gijzelaerr** [comment, marked as answer]
Closing as resolved. The issue was reading 1 byte instead of 4 bytes for REAL values.
